website/docs/index.html.markdown

```diff
@@ -101,6 +101,9 @@ provider "aws" {
 }
 ```
 
+If specifying the profile through the `AWS_PROFILE` environment variable, you
+must also set `AWS_SDK_LOAD_CONFIG` to a truthy value, e.g. `AWS_SDK_LOAD_CONFIG=1`
```

Comment:
The wording of "must" here is overly strong, since `AWS_PROFILE` can successfully be used today in situations without `AWS_SDK_LOAD_CONFIG`. For example, the usage of only the `AWS_PROFILE` environment variable, works when the credentials are in a credentials file and the profile does not have any advanced configurations, such as: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

See also: https://github.com/terraform-providers/terraform-provider-aws/issues/8779#issuecomment-495930164

My recommendation would be to call out specific usages that require the additional environment variable, e.g.

```suggestion
may also need to set `AWS_SDK_LOAD_CONFIG` to a truthy value (e.g. `AWS_SDK_LOAD_CONFIG=1`) for advanced AWS client configurations, such as profiles that use the `source_profile` or `role_arn` configurations.
```